library/spdm_responder_lib/libspdm_rsp_key_exchange.c

```diff
@@ -82,13 +82,16 @@ libspdm_return_t libspdm_get_response_key_exchange(void *context,
                                                0, response_size, response);
     }
 
-    if (!libspdm_is_capabilities_flag_supported(
-            spdm_context, false,
-            0, SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_MEAS_CAP) &&
-        spdm_request->header.param1 > 0) {
-        return libspdm_generate_error_response(
-            spdm_context, SPDM_ERROR_CODE_INVALID_REQUEST,
-            SPDM_KEY_EXCHANGE, response_size, response);
+    if (spdm_request->header.param1 > 0) {
+        if (!libspdm_is_capabilities_flag_supported(
+                spdm_context, false,
+                0, SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_MEAS_CAP) ||
+            (spdm_context->connection_info.algorithm.measurement_spec == 0) ||
+            (spdm_context->connection_info.algorithm.measurement_hash_algo == 0) ) {
```

Comment:
`spdm_context->connection_info.algorithm.measurement_hash_algo` is set to zero if the Responder supports measurements but its `MeasurementHashAlgo` is raw bit stream only (0), correct? In that case it obviously shouldn't return an error.